collector/__init__.py

```diff
@@ -441,36 +432,29 @@ def _process_observation_data(self,
                             wend = progstart
                         # Timing window starts at the beginning of the sequence, the slew can be outside the window
                         # Therefore, subtract acquisition time from start of timing window
-                        wstart -= self.observations[-1].acquisition()
+                        wstart -= self.observations[site][-1].acquisition()
                         windows.append(Time([round_min(wstart), round_min(wend)]))
-                self.obs_windows.append(windows)
+                self.obs_windows[site].append(windows)
 
                 # Observation number in program
                 collected_program.add_observation(self.observation_num)
 
                 # Check if group exists, add if not
-                if group['key'] not in self.scheduling_groups.keys():
-                    self.scheduling_groups[group['key']] = {'name': group['name'], 'idx': []}
+                if group['key'] not in self.scheduling_groups[site].keys():
+                    self.scheduling_groups[site][group['key']] = {'name': group['name'], 'idx': []}
                     collected_program.add_group(group['key'])
-                self.scheduling_groups[group['key']]['idx'].append(Collector.observation_num)
+                self.scheduling_groups[site][group['key']]['idx'].append(Collector.observation_num)
 
                 # Get Horizons coordinates for nonsidereal targets (write file for future use)
                 # if target_tag in ['asteroid', 'comet', 'major-body']: NOTE: This does not work, and it should!
 
-                self.obstatus.append(status)
-                # self.target_name.append(target_name)
-                # self.target_tag.append(target_tag)
-                # self.target_des.append(des)
-                # self.coord.append(SkyCoord(ra, dec, frame='icrs', unit=(u.deg, u.deg)))
-                # self.mags.append(target_mags)
-                self.toostatus.append(too_status)
-                self.priority.append(priority)
-                # self.conditions.append({'iq': condf[0], 'cc': condf[1], 'bg': condf[2], 'wv': condf[3]})
-
-                self.observations.append(Observation(Collector.observation_num,
+                #self.priority.append(priority)
+
+                #logging.debug([oc  for oc in classes if oc != 'ACQ'][0] )
+                self.observations[site].append(Observation(Collector.observation_num,
                                                      obs_odb_id,
                                                      band,
-                                                     Category(obsclass.lower()),
+                                                     ObservationClass(obsclass.lower()),
```

Comment:
I see here. I think we can try to push this back so that `List[ObservationClass]` are being passed to the `Collector` instead of doing the transformation here, but for now, I'll mark this as resolved and investigate it later.